src/odemis/util/transform.py

```diff
@@ -818,3 +818,72 @@ def inverse(self):
         Ainv = numpy.dot(tri_inv(SL), numpy.transpose(self.rotation_matrix))
         tinv = -numpy.dot(Ainv, self.translation)
         return self.__class__(matrix=Ainv, translation=tinv)
+
+
+class AnamorphosisTransform(AffineTransform):
+
+    def __init__(self, params):
+        self.params = params
+
+    @classmethod
+    def from_pointset(cls, x, y):
+        x = numpy.asarray(x)
+        y = numpy.asarray(y)
+
+        w = x[:, 0] + 1.0j * x[:, 1]
+        v = y[:, 0] + 1.0j * y[:, 1]
+
+        # Setup Vandermonde matrix; suffix cc denotes complex conjugate
+        wcc = numpy.conj(w)  # w̄
+        w2 = w * w  # w²
+        wabs2 = w * wcc  # ww̄
+        wcc2 = wcc * wcc  # w̄²
+        w2wcc = w * wabs2  # w²w̄
+        wwcc2 = wcc * wabs2  # ww̄²
+        w3wcc2 = wabs2 * w2wcc  # w³w̄²
+        w2wcc3 = wabs2 * wwcc2  # w²w̄³
+        M = numpy.column_stack((numpy.ones_like(w),  # zero order
+                                w, wcc,  # first order
+                                w2, wabs2, wcc2,  # second order
+                                w2wcc, wwcc2,  # third order
+                                w3wcc2, w2wcc3))  # fifth order
+        params = numpy.linalg.lstsq(M, v)[0]
+        return cls(params=params)
+
+    @property
+    def rotation(self):
+        return self._rotation
+
+    @property
+    def scale(self):
+        return self._scale
+
+    @property
+    def shear(self):
+        return self._shear
+
+    @property
+    def translation(self):
+        return self._translation
+
+    @property
+    def transformation_matrix(self):
+        return self._transformation_matrix
+
+    @property
+    def params(self):
+        return self._params
+
+    @params.setter
+    def params(self, params):
+        self._params = params
+        a, b1, b2 = params[:3]
+        p, q = (b1.real, b1.imag)
+        r, s = (b2.real, b2.imag)
+        matrix = numpy.array([(p + r, -q + s), (q + s, p - r)])
+        R, S = qrp(matrix)
+        self._rotation = _rotation_matrix_to_angle(R)
+        self._scale = numpy.diag(S)
+        self._shear = S[0, 1] / S[0, 0]
```

Comment:
All these lines could be avoided by just using the setter from AffineTransform: `self.transformation_matrix = matrix` (and you'd just need to drop the property from this class).

Reply:
I reworked the code. It now separately handles the linear (affine) and non-linear coefficients of the transform. For all linear coefficients it now uses the functions inherited from AffineTransform. As a bonus, the attributes (properties) are no longer read-only.